web/src/engine/websites/decorators/CoreView.ts

```diff
@@ -108,42 +114,77 @@ export function MangasMultiPageCSS(paths = mangaPaths, query = queryMangas, quer
+
+    const { html } = await FetchJSON<APIChaptersHTML>(new Request(new URL(`/api/viewer/readable_products?aggregate_id=${id}&number_since=99999&number_until=0&read_more_num=150&type=episode`, this.URI)));
+    const doc = new DOMParser().parseFromString(html, 'text/html');
+    return [...doc.querySelectorAll(queryChapters)].map(chapter => {
+        const chapterData = extractor.call(this, chapter);
+        const title = chapterData.title.replace(manga.Title, '').trim();
+        return new Chapter(this, manga, chapterData.id, !title ? chapterData.title: title);
     });
 }
 
 /**
- * A class decorator that adds the ability to extract all chapters for a given manga from this website using the given CSS {@link query}.
- * The chapters are extracted from the composed url based on the `Identifier` of the manga and the `URI` of the website.
- * @param query - A CSS query to locate the elements from which the chapter identifier and title shall be extracted
- * @param extractor - An Extractor to get chapter infos
+ * An extension method for extracting chapters using Coreview API. Use this when endpoint 'pagination_readable_products' is available. Multipaged.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
  */
-export function ChaptersSinglePageCSS(query: string = queryChapters, extractor = ChapterExtractor) {
+export function ChaptersMultiPageAJAXV2() {
     return function DecorateClass<T extends Common.Constructor>(ctor: T, context?: ClassDecoratorContext): T {
         Common.ThrowOnUnsupportedDecoratorContext(context);
         return class extends ctor {
             public async FetchChapters(this: MangaScraper, manga: Manga): Promise<Chapter[]> {
-                return FetchChaptersSinglePageCSS.call(this, manga, query, extractor);
+                return FetchChaptersMultiPageAJAXV2.call(this, manga);
             }
         };
     };
 }
 
+/**
+ * A class deocrator for extracting chapters using Coreview API. Use this when endpoint 'pagination_readable_products' is available. Multipaged.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
+ * @param manga - A reference to the {@link Manga} which shall be assigned as parent for the extracted chapters
+ */
+export async function FetchChaptersMultiPageAJAXV2(this: MangaScraper, manga: Manga): Promise<Chapter[]> {
+    const jsonData = (await FetchCSS(new Request(new URL(manga.Identifier, this.URI)), 'script#episode-json')).shift().dataset.value;
+    const { readableProduct: { series: { id } } } = JSON.parse(jsonData) as JSONSerie;
+    const chapterList: Chapter[] = [];
+    for (let offset = 0, run = true; run;) {
+        const data = await FetchJSON<APIChapter[]>(new Request(new URL(`/api/viewer/pagination_readable_products?type=episode&aggregate_id=${id}&sort_order=desc&offset=${offset}`, this.URI)));
```

Comment:
May construct the `URI` and its search parameters separately for better readability